### ISSUE TITLE
Hammering remarkForm with lots of changes very quickly causes page to crash.

### ISSUE THREAD

**Issue by kcexn** [tinacms/tinacms#763]
## Description

When using the remarkForm HOC. It seems that hammering the form input with changes can sometimes cause the page to crash with the following error:

TypeError: e is undefined

and a very big stack trace (attached).
[StackTrace.txt](https://github.com/tinacms/tinacms/files/4208435/StackTrace.txt)

It seems there have been some other similar issues about this in the past:

[gatsbyJS issue](https://github.com/gatsbyjs/gatsby/issues/11183)
#52 

Both of these have been closed already.

### Steps to reproduce

Clone my repository, which is a blank GatsbyJS hello world repository with an index.js page 
and tinaCMS set up.
`
git clone git@github.com:kcexn/kcexn-learn-gatsby-project-3.git`

start the gatsby development server

```bash
cd kcexn-learn-gatsby-project-3
gatsby develop
```
go to the remarkForm and type very quickly into the input (spam the keyboard)

### Expected result

The webpage should reflect the input in the remarkForm (doesn't need to be instantaneous if changes are occuring too quickly, if that is a constraint right now) without crashing.

### Actual result

The Webpage crashes, need to recover be refreshing the page.

#### Tinacms Package Versions

```text
npm ERR! peer dep missing: typescript@>=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev || >= 3.4.0-dev || >= 3.5.0-dev || >= 3.6.0-dev || >= 3.6.0-beta || >= 3.7.0-dev || >= 3.7.0-beta, required by tsutils@3.17.1
npm ERR! peer dep missing: typescript@>=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev || >= 3.4.0-dev || >= 3.5.0-dev || >= 3.6.0-dev || >= 3.6.0-beta || >= 3.7.0-dev || >= 3.7.0-beta, required by tsutils@3.17.1
npm ERR! peer dep missing: prop-types@>15, required by @tinacms/form-builder@0.2.14
npm ERR! peer dep missing: prosemirror-tables@^0.9.1, required by prosemirror-utils@0.9.6
npm ERR! peer dep missing: typescript@>=3.7.0, required by ts-essentials@5.0.0
├─┬ gatsby-plugin-tinacms@0.1.23
│ └─┬ tinacms@0.13.0
│   ├── @tinacms/core@0.7.1
│   ├─┬ @tinacms/fields@0.4.1
│   ├─┬ @tinacms/form-builder@0.2.14
│   ├─┬ @tinacms/forms@0.1.3
│   ├── @tinacms/icons@0.5.1
│   ├─┬ @tinacms/react-core@0.2.6
│   ├── @tinacms/styles@0.1.2
├─┬ gatsby-tinacms-git@0.2.19
│ ├─┬ @tinacms/api-git@0.5.5
│ └── @tinacms/git-client@0.4.2
├─┬ gatsby-tinacms-remark@0.7.2
```

#### Tools

| Tool    | Version |
| ------- | ------- |
| browser |    Firefox 72.0.2     |
| os      |    Debian 10.2     |
| node    |    v12.16.0     |
| npm     |    6.13.7     |

**Comment by ncphillips** [comment]
This is essentially a duplicate of #52.

We tried to fix this by introducing a 1000 ms debounce to file writign, but that's doesn't always seem to be enough. 

@PaulBunker submitted #817 that makes that debounce time configurable. Hopefully this help with this issue.

It will be available on Monday

**Comment by ncphillips** [comment]
Hey @kcexn did you get a chance to see if setting the [TINA_GIT_DEBOUNCE_MS](https://tinacms.org/docs/gatsby/configure-git-plugin/#configuring-the-file-writing-debounce) environment variable helped address this issue?

**Comment by kcexn** [comment]
Hi @ncphillips I took a look at the environment variable. I tried changing it to 3000 and to 100 and it didn't resolve the issue. I also tried it in my Arch VM and added some different network latencies to see if that made a difference and it didn't seem to.

It's a little difficult to tell since it happens a little bit randomly.

What did end up seeming to fix the problem though was adding a dummy file as described at the bottom of [your tutorial](https://tinacms.org/docs/gatsby/markdown).

Even though in my extremely trivial exercise there is never an empty field. I am not editing any of the fields in the front matter.

Actually this whole thing is making me go a bit bananas. I run the development server with:
```TINA_GIT_DEBOUNCE_MS=3000 gatsby develop -H 0.0.0.0```

then when I make a modification in the remarkForm field. I expected it to have a 3 second delay before the preview updates, but it still happens pretty quickly.
